### PR TITLE
Asset characteristics in property view

### PR DESCRIPTION
**Pull request by adamtry**
# [POH-46](https://hackney.atlassian.net/jira/software/projects/POH/boards/100?selectedIssue=POH-46)
> As a Manage My Home User I need to to see more property information, to make informed decisions

image

## What
Added an expandable element on the property page to display asset characteristics from the asset API response

## Why
There's been a request to display this information and these particular fields have been provided and confirmed.

## How
- Pull the assetCharacteristics from the API response and pass them to the new element in the property page sidebar
- Use the data model for assetCharacteristics from the Common MFE
- Sort the display so that all are displayed but non-null fields are on top

[POH-46]: https://hackney.atlassian.net/browse/POH-46?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/asset-characteristics/asset-characteristics.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import React from "react";
+
+import { mockAssetV1, render } from "@hackney/mtfh-test-utils";
+import { screen } from "@testing-library/react";
+
+import { locale } from "../../services";
+import { PropertySpecification } from "./asset-characteristics";
+
+describe("AssetCharacteristics", () => {
```
Comment (humulla) [inline src/components/asset-characteristics/asset-characteristics.spec.tsx:9]: Maybe you should add another test to not display the characteristics if the value is null

File: src/views/asset-view/layout.tsx [hunk 2]
```diff
@@ -56,6 +61,14 @@ const AssetSideBar = ({
...
             assetReference={assetId}
           />
+          <details className="govuk-details" data-module="govuk-details">
+            <summary className="govuk-details__summary">
+              <span className="govuk-details__summary-text">Property Specification</span>
+            </summary>
+            <div className="govuk-details__text">
+              <PropertySpecification assetCharacteristics={assetCharacteristics} />
+            </div>
+          </details>
```
Comment (Duslerke) [inline src/views/asset-view/layout.tsx:71]: This looks like something that ought to belong within the new component, or am I mistaken?

File: src/components/asset-characteristics/asset-characteristics.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  // Loop through props as SummaryListItems
+  return (
+    <aside className="mtfh-asset-charateristics">
+      <SummaryList overrides={[0.5, 0.5]} variant="base">
+        {Object.keys(propSpecProps)
+          // Sort the labels not null on top
+          .sort((label) => {
+            if (propSpecProps[label].value == null) return 1;
+            return -1;
+          })
```
Comment (Duslerke) [inline src/components/asset-characteristics/asset-characteristics.tsx:83]: I know it's deliberate... I'm personally not too fond of sorting as I feel that a rigid information display structure gives familiarity & ease of use going forward because "as the user I don't have to guess where within the asset characteristics list, my wanted piece of data is".

But that's my opinion... I can also see the "clean look/easier to read" argument behind the "sort" decision. I'd say these are close to equal. I'd probably want to brush this design through a user or anyone non-dev really. Pretty much ask: "Do you like the screenshot of non-sorted list, or sorted list more?".

**Comment by Duslerke** [comment]
Btw, @adamtry I saw your post of not having access to SC. So I'll drop this here...

**The SC is failing because of:**
image

I saw you've added at least 1 test, so it shouldn't show 0%. Chances are the SC is misconfigured like it used to be on the finance APIs and listeners. Pretty much, it's not receiving a codecov report, so it's making a safe assumption of 0%.

What's odd is that... looking back at the last commit on main branch... [the sonar didn't pick up any codecov there either](https://github.com/LBHackney-IT/mtfh-frontend-property/runs/14186985882), but it didn't seem to complain. @humulla Did we stricten the Quality Gate in any way?

File: src/components/asset-characteristics/asset-characteristics.tsx (hunk 3 above)
Comment (Duslerke) [reply src/components/asset-characteristics/asset-characteristics.tsx:83]: **TLDR;**
I think data fields jumping up & down in position going from one record to another will end up being more of a nuisance than a help. It may be good to drop the sorting/not-sorting decision onto the users so they make the choice themselves.

Comment (adamtry) [reply src/components/asset-characteristics/asset-characteristics.tsx]: Sorting removed, as confirmed by Sean, thanks for the heads-up.

**Review by humulla — APPROVED** [review]

**Comment by Duslerke** [comment]
Sonar fails due to "no **detected** codecov". This is because the setup for this is incomplete. It's separate task that will be tackled by a dedicated tech-debt ticket. For now we'll have to live with the red X

**Comment by Duslerke** [comment]
Had to make the merge commit as Callum has disabled a few features for their release.
